apps/frontend/src/lib/storage/localStorage/user/connection/message/index.ts

```diff
@@ -148,6 +155,23 @@ async function handleEdges(session: Session, user: User, messages: EdgeMessage[]
   return newBackupEntries;
 }
 
+async function handleConnectionPSIs(session: Session, user: User, messages: PSIMessage[]): Promise<void> {
+  for (const message of messages) {
+    if (message.senderUsername && user.connections && user.connections[message.senderUsername]) {
+
+      if (user.userData?.settings?.automaticPSIEnabled) {
+        // Only if the setting is enabled should PSI be automatically refreshed
+        const newVerifiedIntersection = await refreshPSI(user, user.connections[message.senderUsername]);
+
+        if (newVerifiedIntersection) {
+          // If intersection is successful, set the size of intersection
+          await updateConnectionPSISize(newVerifiedIntersection, user, session, user.connections[message.senderUsername], true);
```

Comment:
True is passed in here to set refreshPSI to true, so that on the connection page we'll fetch the results from the PSI.